swagger.yaml

```diff
@@ -49,6 +74,22 @@ paths:
           description: "Recipe not found"
         500:
           $ref: '#/responses/InternalError'
+    put:
+      summary: "Update a recipe"
+      description: |
+        Update a recipe by getting the specific recipe by id and update in mongo
+      parameters:
+      - $ref: '#/parameters/recipe_id' 
+      - $ref: '#/parameters/new_recipe'
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "A json object of a recipe which has been updated"
+          schema:
+            $ref: '#/definitions/Recipe'
+        500:
```

Comment:
Likely going to be 400 status code for bad request, again need to define what fields are mandatory and which ones are not

Reply:
As part of the validation, a 400 status code for bad request should arise when any field of the recipe is `nil`

Reply:
fyi - You will need to think of other validation, not just nil. For example empty strings etc.

Reply:
Resolved in the last commits:

_- Add simplified validation which follows model structure and change isHierarchy from bool to *bool to support validation

- Simplify UpdateRecipeFieldsFromPrevious to avoid using 'reflect' and use if statement to satisfy model structure_